src/platform/packages/shared/shared-ux/code_editor/impl/mods/components/widgets_customizations.tsx

```diff
@@ -34,34 +34,32 @@ export const EditorWidgetsCustomizations: FC<
 > = ({ children, editor, enableSuggestWidgetRepositioning }) => {
   const { euiTheme } = useEuiTheme();
   const [globalStyleModifierClassNamesRecord, setGlobalStyleModifierClassNamesRecord] = useState<
-    GlobalProps['styles']
+    Extract<GlobalProps['styles'], CSSObject>
   >({});
   const headerOffset = useRef(
     'var(--kbn-layout--application-top, var(--euiFixedHeadersOffset, 0px))'
   );
 
   /**
-   * @description Repositioning handlers for widgets of interest, only for widgets known to support the "onDidShow" and "onDidHide" events,
+   * @description Repositioning handlers for widgets of interest, assumes the widget has the the "onDidShow" and "onDidHide" methods,
    * See {@link https://github.com/elastic/kibana/issues/223981 | GitHub issue #223981} for the rationale behind this bug fix implementation
    */
   const registerRepositionHandlers = useCallback(
     (
-      widget: monaco.editor.IContribution['widget'] | undefined,
-      domNodeGetter: () => HTMLElement | null,
+      widget: monaco.editor.IContentWidget | undefined,
+      domNodeGetter: (widget: monaco.editor.IContentWidget) => HTMLElement | null,
       modifierClassName: string
     ) => {
-      // The "onDidShow" and "onDidHide" is not documented so we guard from possible changes in the underlying lib
+      // @ts-expect-error -- The "onDidShow" and "onDidHide" is not documented so we guard from possible changes in the underlying lib
       if (widget && widget.onDidShow && widget.onDidHide && enableSuggestWidgetRepositioning) {
         let $widgetNode: HTMLElement | null = null;
 
         // Add the modifier className to the global style modifier class names record
-        setGlobalStyleModifierClassNamesRecord((prev) => ({
-          ...prev,
+        setGlobalStyleModifierClassNamesRecord((prevStyles) => ({
+          ...prevStyles,
           [`.${modifierClassName}`]: {
-            // @ts-expect-error -- the handling of showing and hiding of the widgets by monaco
-            // happens through style attributes hence the need for the important modifier
             visibility: 'hidden !important',
-          },
+          } as unknown as CSSObject,
```

Comment:
We type cast this to `CSSObject` because the string `hidden !important` is not a valid value for `visibility`, however we need the important modifier because monaco applies it's own styling leveraging style attributes which would have a higher specificity than the declaration from our className.